src/mod_pushoff.erl

```diff
@@ -415,14 +443,27 @@ backend_configs(Host) ->
 
 -spec(start_worker(Host :: binary(), Backend :: backend_config()) -> ok).
 
-start_worker(Host, #backend_config{type = Type, certfile = CertFile, gateway = Gateway}) ->
-    Module = proplists:get_value(Type, [{apns, ?MODULE_APNS}]),
+start_worker(Host, #backend_config{type = Type, certfile = CertFile, gateway = Gateway, api_key = ApiKey}) ->
```

Comment:
you could split `#backend_config` into two records, one for fcm and one for apns and avoid stacking different sets of values into one record